tools/ddrgen/src/scanners/dwarf/DwarfScanner.cpp

```diff
@@ -191,7 +193,10 @@ DwarfScanner::blackListedDie(Dwarf_Die die, bool *dieBlackListed)
 		/* If the decl_file matches an entry in the file table not beginning with "/usr/",
 		 * then we are interested in it. Also filter out decl file "<built-in>".
 		 */
-		if ((NULL == strstr(_fileNamesTable[declFile - 1], "<built-in>"))
+		if (0 == declFile) {
+			/* What happens when there is no file? */
```

Comment:
Re "What happens when there is no file?"
If a Die has no decl_file attribute, then it will typically not have the decl_file attribute and the above dwarf_hasattr() check will return false. Its strange that it would crash here now but did not before. I guess there must be some odd case where it uses 0 to indicate no file.